feat: VS Code plugin open PR list ALA-697 (#1936)
codacy/docs: docs/getting-started/integrating-codacy-with-visual-studio-code.md

!!! info "To use this extension you need a [Codacy account](https://www.codacy.com/signup-codacy)"

## Interface overview

> The main view of the extension displays information about the code quality changes introduced by the pull request you're working on, as well as the status of other open pull requests and analyzed branches. This information is grouped into three tabs:

Author: Good point. Section works, yes.

```suggestion
The main view of the extension displays information about the code quality changes introduced by the pull request you're working on, as well as the status of other open pull requests and analyzed branches. This information is grouped into three sections:
```

-   [Status](#status-tab)
-   [Open Pull Requests](#open-pull-requests-tab)
-   [Analyzed branch](#analyzed-branch-tab)

![Important issues](./images/codacy-vscode-extension-main-view.png)

### Status tab

> The pull request **Status** tab displays the following information for the currently checked out pull request:

Author: I initially used something similar to what you suggested, but GitHub seems OK with using "check out a PR" on [their docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally), and I liked the idea of shortening the sentence.

If you think it may confuse our users or find it confusing yourself, I'm happy to use your variant.

> The pull request **Status** tab displays the following information for the currently checked out pull request:

Reviewer: For users like me who might not be completely comfortable with git operations, I would prefer to be technically accurate. 🤔 
If checking out a branch is different from checking out a pull request, let's use the correct term for the operation Codacy is executing. On the other hand, if it's technically the same, then I believe it's ok to use either.
Thoughts?

> The **Pull request status** tab displays the following information for the currently checked out pull request:

Author: ```suggestion
The **Pull request status** tab displays the following information for the pull request of the currently checked-out branch:
```

> ![Codacy main view](images/codacy-vscode-extension-status-tab.png)

Author: Yes, I was on the fence about this one too. I'm not a fan of starting a section with an image, but since these are small, we can.

-   Any **Quality issues** introduced or fixed by the pull request. These are the same issues you find in the [Quality Issues tabs](../repositories/pull-requests.md#issues-tabs) in the Codacy app and are also visible in VS Code's Problems tab.

>     The number next to each file name is the total number of Quality issues that the pull request adds to or removes from that file. The number farther to the right is the number of new Medium or Critical Quality issues. If there are any such issues, the file name is also highlighted in yellow (Medium) or red (Critical).

Reviewer 2: The number farther to the right is actually added by VSCode, and it's the amount of problems on that file. Problems that may or may not come from Codacy.

Author: Done 👍

## Getting pull request quality and coverage data

> To see Codacy quality and coverage data for a pull request, follow these steps:

Reviewer: When we had "the pull request you're working on", it was implicit they were open. Removing that expression, it's not so clear anymore... 
Again for the sake of technical accuracy, I would keep "**open** pull request". And also because it's short, it doesn't increase the sentence complexity, IMO.

1.  Open the repository directory in VS Code.

    !!! note
        If the repository isn't on Codacy yet, [add it to Codacy](../organizations/managing-repositories.md#adding-a-repository) first.

> 1.  Open the main view by clicking the **Codacy logo** in the activity bar or the **Codacy tab** in the status bar.

Reviewer: If we refer to the **Codacy tab** in the status bar, we should include it on the screenshot. 🤔

Author: Done!

> 1.  Open the main view by clicking the **Codacy logo** in the activity bar.

Author: It's visible sometimes, but not consistently. I opted to remove it; open to reverting this.

@alerizzo thoughts?

> 1.  Open the main view by clicking the **Codacy logo** in the activity bar.

Reviewer 2: See no reason for this change either. Still can be opened with both the left bar or the bottom bar.

> 1.  Open the main view by clicking the **Codacy logo** in the activity bar.

Author: ```suggestion
1.  Open the main view by clicking the **Codacy logo** in the activity bar or the **Codacy tab** in the status bar.
```

    ![Codacy main view](images/codacy-vscode-extension-sign-in.png)

1.  If you’re not signed in, click the **Sign in** button to authorize VS Code on Codacy.

    !!! tip

>         You can alternatively sign in, and access more Codacy commands, such as **Sign out**, by opening the VS Code command palette (`Ctrl+Shift+P` on Windows/Linux or `Cmd+Shift+P` on macOS) and typing "Codacy".

Author: That's OK, but I'd still keep the "value" of the action first and the "how" later. How about this? I further simplified:

```suggestion
        To access a complete list of Codacy commands, open the VS Code command palette (`Ctrl+Shift+P` on Windows/Linux or `Cmd+Shift+P` on macOS) and type "Codacy".
```

> 1.  Checkout the pull request of interest, either manually or from the **Open Pull Requests** tab, by clicking the arrow button or using the contextual right-click menu.

Reviewer: Tweaks:
- Breaking the sentence, for clarity.
- Changing the check-out object to "branch". Please evaluate. 🙏 

```suggestion
1.  Check out the branch associated with the pull request of interest. You can do it either manually or from the **Open Pull Requests** tab, by clicking the arrow button or using the contextual right-click menu.
```

Author: OK to break up the sentence, with two tweaks:

* I'd still refer to a "pull request", see my previous comment
* It seems that "checkout" is also commonly used. It's a subtle difference, and both are considered valid in different contexts. We can use "check out" here since we're referring to a PR. But if we decide to refer to a branch, "checkout" would make more sense IMO, as it's closer to the Git command.

```suggestion
1.  Check out the pull request of interest. You can do it either manually or from the **Open Pull Requests** tab, by clicking the arrow button or using the contextual right-click menu.
```
